### ISSUE TITLE
Properties are considered as methods

### ISSUE THREAD

**Issue by tristanlatr** [twisted/pydoctor#233]
Currently `@property` decorator is not understood correctly.
It shows as a method, but is should show as a instance variable.

The docstring should be extracted from the getter (not sure if it's already the case)

This is a bug report.

The workaround is to create the property with the `property(fset, fget)`, then it's understood.

**Comment by mthuurne** [comment]
Agreed.

I think currently the docstring from the getter is shown simply because the getter is presented as a method.

If the setter has a docstring as well, it might be worth combining it with the getter. A non-trivial setter might want to document which exception types it can raise if an attempt to assign an incorrect value is made.

Note that if the method is decorated with both `@property` and `@classmethod`, it should be listed as a class variable.

It would be useful to report in the output whether a property is read-only or read/write (whether it has a setter). And it would be useful to show properties being read-only for frozen `attrs` attributes ([`attr.setters.frozen()`](https://www.attrs.org/en/stable/api.html#attr.setters.frozen) or frozen classes) as well.

**Comment by tristanlatr** [comment]
> The workaround is to create the property with the property(fset, fget)

Whoooops I'm unsure why, but it seems that it's not working all the time with the property(fget, fset) function.
I.e here: https://mfesiem.github.io/docs/msiempy/dev/msiempy.core.types.NitroDict.html#text

Where `text` and `json` should be considered as instance variables.

**Comment by mthuurne** [comment]
There is no support for `property(fget, fset)` in the code at all, so that it even somewhat works is just a coincidence...

I am currently trying to add support for the `@property` decorator. The most simple use case is now working fine, but I'm not sure yet how many non-simple cases there will be and whether it is a good idea to do this before the release.

**Comment by adiroiban** [comment]
For a property we have: getter, setter and deleter :)
I can't think of an use case in which you want extra documentation for the deleter. 

In my code, to keep it simple, I am only documenting the getter.

I guess that for pydoctor it would be intersting to mark the property as read-only, if a setter is not defined.

**Comment by mthuurne** [comment]
I think that in the future we probably want to revisit how properties are presented and create some kind of unified display that combines information from the getter, setter and deleter in a single output box. But that's too ambitious to do right now.

(Copy-pasted from the PR, since it has a better chance of being found here.)

**Comment by tristanlatr** [comment]
Hello @mthuurne , I think there is an issue with the current implementation: `@abc.abstractproperty` is not rendering as a property but still as a method. 

We can see this issue on the new `Template` object I'm working on: https://pydoctor--299.org.readthedocs.build/en/299/api/pydoctor.templatewriter.Template.html#version

**Comment by mthuurne** [comment]
Indeed `@abc.abstractproperty` is not recognized correctly. The problem is in the `abc.` part, so you can work around it like this:
```py
    from abc import abstractproperty
    class C:
        @abstractproperty
        def version(self) -> int:
            pass
```
This should be fairly simple to fix; I might have a go at it tomorrow.

**Comment by mthuurne** [comment]
Fixed in #344.
